File: identity_matrix/identity.c
#include <stdio.h>

void identify(int vector[][4], int length){

    for(int i = 0; i < length; i++){
        for(int j = 0; j < length; j++){
            if((i == j && vector[i][j] != 1) || (i != j && vector[i][j] != 0)){
                printf("is not a identity matrix");
                return;
            }
        }
    }
    printf("is a identity matrix");
}

int main(){
    int vector[4][4]= {
        {1, 0, 1, 0},
        {0, 1, 0, 0},
        {0, 0, 1, 0},
        {0, 0, 0, 1}
    };

    int length = sizeof(vector) / sizeof(vector[0]); 

    identify(vector, length);

    return 0;
}

File: identity_matrix/identity.js
const verify = (matrix) => {
    const size = matrix.length 
    for(let i = 0; i < size; i++){
        for(let j = 0; j < size; j++){
            if((i == j && matrix[i][j] != 1) || (i != j && matrix[i][j] != 0)){
                console.log(`is not a matrix identity`)
                return
            }
        }
    }
    console.log(`is a matrix identity`)
}

testMatrix = [ 
    [1, 0, 0, 0],
    [0, 1, 0, 0],
    [0, 0, 1, 0],
    [0, 0, 0, 1]
]

verify(testMatrix)